checker/src/org/checkerframework/checker/index/upperbound/UpperBoundTransfer.java

```diff
@@ -204,6 +205,35 @@ private UBQualifier accountForLowerBoundAnnos(Node node, UBQualifier qual, Strin
         return qual;
     }
 
+    private void propagateToSubterm(
+            Node target, Node offset, UBQualifier qualifier, CFStore store, boolean plusNode) {
+        UBQualifier existingQual =
+                UBQualifier.createUBQualifier(
+                        atypeFactory
+                                .getAnnotatedType(target.getTree())
+                                .getAnnotationInHierarchy(atypeFactory.UNKNOWN));
+
+        UBQualifier newQual =
+                existingQual.glb(
+                        plusNode
+                                ? qualifier.plusOffset(offset, atypeFactory)
+                                : qualifier.minusOffset(offset, atypeFactory));
+
+        Receiver rightRec = FlowExpressions.internalReprOf(analysis.getTypeFactory(), target);
+        store.insertValue(rightRec, atypeFactory.convertUBQualifierToAnnotation(newQual));
+    }
+
+    private void propagateToSubterms(
+            BinaryOperationNode node, UBQualifier qualifier, CFStore store) {
+        boolean plusNode = node instanceof NumericalAdditionNode;
+        // Otherwise this should be a numerical subtraction node.
+
+        propagateToSubterm(
+                node.getRightOperand(), node.getLeftOperand(), qualifier, store, plusNode);
+        propagateToSubterm(
```

Comment:
I dug a little deeper and the result of this code:

```java
    public void subtraction5(String[] a, int i) {
	if (1 - i < a.length) {
	    @IndexFor("a") int j = i;
	}
    }
```
is this warning:

```
ParserOffsetTest.java:58: error: [assignment.type.incompatible] incompatible types in assignment.
	    @IndexFor("a") int j = i;
	                           ^
  found   : @LTEqLengthOf("a") int
  required: @LTLengthOf(value="a", offset={}) int
```
which is still wrong: `i` is definitely not `@LTEqLengthOf("a")`